nautobot/dcim/api/serializers.py

```diff
@@ -303,6 +303,13 @@ class LocationSerializer(NautobotModelSerializer, TaggedModelSerializerMixin, St
     tenant = NestedTenantSerializer(required=False, allow_null=True)
     site = NestedSiteSerializer(required=False, allow_null=True)
     tree_depth = serializers.SerializerMethodField(read_only=True)
+    time_zone = TimeZoneSerializerField(required=False, allow_null=True)
+    circuit_count = serializers.IntegerField(read_only=True)
+    device_count = serializers.IntegerField(read_only=True)
+    prefix_count = serializers.IntegerField(read_only=True)
+    rack_count = serializers.IntegerField(read_only=True)
+    virtualmachine_count = serializers.IntegerField(read_only=True)
+    vlan_count = serializers.IntegerField(read_only=True)
```

Comment:
Since these aren't model fields, you probably need to update `LocationViewSet.queryset` to include the appropriate `annotate(count_related(...), ...)` calls, similar to what's on `SiteViewSet.queryset` today.